app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -520,6 +529,81 @@ private void showInfoDialog() {
+        currentTextview.setPadding(20, 0, 30, 0);
+        currentTextview.setTextSize(settings.getFontSizeMax(settings.getSmallFont()));
+        currentTextview.setTextColor(Color.GRAY);
+        currentTextview.setAutoLinkMask(Linkify.ALL);
+        currentTextview.setText(getString(R.string.currentBalanceSentence, Utils.formatBalance(this, currentBalance, loyaltyCard.balanceType)));
+        textLayout.addView(currentTextview);
+
+        TextView updateTextview = new TextView(this);
+        updateTextview.setPadding(30, 0, 20, 0);
+        updateTextview.setTextSize(settings.getFontSizeMax(settings.getSmallFont()));
+        updateTextview.setTextColor(Color.GRAY);
+        updateTextview.setAutoLinkMask(Linkify.ALL);
+        updateTextview.setText(getString(R.string.newBalanceSentence, Utils.formatBalance(this, currentBalance, loyaltyCard.balanceType)));
+        textLayout.addView(updateTextview);
+
+        layout.addView(textLayout);
+
+        final EditText input = new EditText(this);
+        input.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_FLAG_DECIMAL);
+        input.setHint(R.string.updateBalanceHint);
+        input.addTextChangedListener(new SimpleTextWatcher() {
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                updateTextview.setText(getString(R.string.newBalanceSentence, Utils.formatBalance(dialogContext, calculateNewBalance(s), loyaltyCard.balanceType)));
+            }
+        });
+        layout.addView(input);
+
+        updateDialog.setView(layout);
+        updateDialog.setPositiveButton(R.string.ok, (dialogInterface, i) -> {
+            dialogInterface.dismiss();
+            handleBalanceUpdate(input.getText());
+        });
+        updateDialog.setNegativeButton(getString(R.string.cancel), (dialogInterface, i) -> dialogInterface.cancel());
+        updateDialog.create().show();
+    }
+
+    private BigDecimal calculateNewBalance(CharSequence s){
+        BigDecimal current = currentBalance;
+        BigDecimal newBalance = current;
+        try {
+            BigDecimal value = Utils.parseBalance(s.toString(), loyaltyCard.balanceType);
+            newBalance = current.subtract(value).max(new BigDecimal(0));
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        return newBalance;
+    }
+
+    private void handleBalanceUpdate(Editable text){
+        BigDecimal newBalance = calculateNewBalance(text);
+        currentBalance = newBalance;
+        SQLiteDatabase mDatabase = new DBHelper(this).getWritableDatabase();
+        DBHelper.updateLoyaltyCard(mDatabase, loyaltyCardId, loyaltyCard.store, loyaltyCard.note, loyaltyCard.expiry, currentBalance, loyaltyCard.balanceType, loyaltyCard.cardId, loyaltyCard.barcodeId, loyaltyCard.barcodeType, loyaltyCard.headerColor, loyaltyCard.starStatus, null, loyaltyCard.archiveStatus);
+    }
```

Comment:
I noticed that setting the balance to 0 doesn't make the icon disappear even though there is no way to "spend more of it".

Probably need to call the `setBottomAppBarButtonState` function here

Reply:
Good catch! Thanks!